core/domain/exp_services_test.py

```diff
@@ -1305,6 +1305,49 @@ def test_unpublished_explorations_can_accept_commit_message(self):
             ), None)
 
 
+class UpdateExplorationContributionDatetimeTests(ExplorationServicesUnitTests):
+    """Test whether contribution date changes with publication of exploration
+    and update of already published exploration
+    """
+
+    def setUp(self):
```

Comment:
What if A and B both make commits to an exploration and then later B publishes it? Does A's contribution count?